.github/actions/buildbook/action.yml

```diff
@@ -11,13 +11,13 @@ inputs:
   jb-save:
     description: "Save the Jupyterbook Build (boolean)"
     required: true
+  token:
+    description: 'A GitHub Personal Access Token (for publishing)'
+    required: false
 
 runs:
   using: "composite"
   steps:
-    - name: Checkout repository
-      uses: actions/checkout@v2
-
```

Comment:
Any idea why we can't do the checkout as part of the composite action? It seems to defeat the purpose to some extent that we had to put it back into each action as per #42.

Reply:
You have to run checkout before using composite actions b/c otherwise the GitHub runner machine does not have anything in the repository. Error message from logs: `Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under '/home/runner/work/jupyterbook-template/jupyterbook-template/.github/actions/buildbook'. Did you forget to run actions/checkout before running your local action?`

Reply:
Okay - I see. It looks like it [can be done](https://manzoorwani.dev/how-to-use-the-new-composite-github-actions-for-reusability/), but not easily with actions stored within the repo they're running on (at least for a template repo where the repo name will change every time it's used!).

Reply:
I see this `action.yml` as a good unit that maintains all the steps to build a book. Not having the checkout in there seems acceptable and almost like a ‘chicken and egg’ problem.